Utilities/Mergers/test/customTopologyCommon.h

```diff
@@ -0,0 +1,184 @@
+        })}};
+      specs.push_back(producer);
+    }
+
+    return inputs;
+  }
+
+  void generateMergers(WorkflowSpec& specs, const Inputs& mergerInputs, mergers::InputObjectsTimespan mergerType)
+  {
+    using namespace mergers;
+
+    MergerInfrastructureBuilder mergersBuilder;
+    mergersBuilder.setInfrastructureName("custom");
+    mergersBuilder.setInputSpecs(mergerInputs);
+    mergersBuilder.setOutputSpec({{"main"}, origin, description, 0});
+
+    MergerConfig config;
+    config.inputObjectTimespan = {mergerType};
+    std::vector<std::pair<size_t, size_t>> param = {{5, 1}};
+    config.publicationDecision = {PublicationDecision::EachNSeconds, param};
+    config.mergedObjectTimespan = {MergedObjectTimespan::FullHistory};
+    config.topologySize = {TopologySize::NumberOfLayers, 2};
+
+    if (mergerType != mergers::InputObjectsTimespan::FullHistory) {
+      mergersBuilder.setOutputSpecMovingWindow({{"main"}, origin, description_moving_window, 0});
+      config.publishMovingWindow = {PublishMovingWindow::Yes};
+    }
+
+    mergersBuilder.setConfig(config);
+
+    mergersBuilder.generateInfrastructure(specs);
+  }
+
+  void generateCheckerIntegrating(WorkflowSpec& specs)
+  {
+    specs.push_back(DataProcessorSpec{
+      "data-checker",
+      Inputs{
+        {"custom", origin, description, 0, Lifetime::Sporadic},
+        {"custom_mw", origin, description_moving_window, 0, Lifetime::Sporadic},
+      },
+      Outputs{},
+      AlgorithmSpec{
+        AlgorithmSpec::InitCallback{[expectedResult = mExpectedResult](InitContext&) {
+          return AlgorithmSpec::ProcessCallback{
+            [expectedResult, numberOfCalls = 0, numberOfObjects = 0, numberOfMovingWindows = 0, lastObjectValue = 0](ProcessingContext& processingContext) mutable {
+              numberOfCalls++;
+
+              if (processingContext.inputs().isValid("custom")) {
+                auto obj = processingContext.inputs().get<mergers::CustomMergeableObject*>("custom");
+                numberOfObjects++;
+                // we are keeping only the last value of secret, as there can be inconsitencies caused
+                // by the lack of synchronisation between merger layers
+                lastObjectValue = obj->getSecret();
+              }
+
+              if (processingContext.inputs().isValid("custom_mw")) {
+                auto mw = processingContext.inputs().get<mergers::CustomMergeableObject*>("custom_mw");
+                numberOfMovingWindows++;
+                // it is not possible to check for correct value in moving window as the value can be lost due
+                // to the sync between layers of mergers and movement of the window
+              }
+
+              if (numberOfCalls == 5) {
+                processingContext.services().get<ControlService>().readyToQuit(QuitRequest::All);
+
+                if (numberOfObjects != 5 || numberOfMovingWindows == 0 || numberOfMovingWindows >= 3) {
```

Comment:
Please add a word or two why `5` was chosen.

Also, I would be more relaxed with the number moving windows expected. Since the computing system is asynchronous by design, one could see more extreme variations in an overloaded CI machine. Perhaps just more than zero and less than 10?